bip44/src/lib.rs

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************************
+*   (c) 2020 ZondaX GmbH
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+********************************************************************************/
+use byteorder::{LittleEndian, WriteBytesExt};
+
+pub mod errors;
+
+use crate::errors::BIP44PathError;
+
+const HARDENED_BIT: u32 = 1 << 31;
+
+#[derive(Debug)]
+pub struct BIP44Path(pub [u32; 5]);
+
+/// Bip44Path
+///
+/// Implementation of the BIP44 standard for derivation path.
+///
+impl BIP44Path {
+    pub fn from_slice(path: &[u32]) -> Result<BIP44Path, BIP44PathError> {
+        let mut path_array: [u32; 5] = Default::default();
+        if path.len() != 5 {
+            return Err(BIP44PathError::InvalidLength);
+        };
+
+        path_array.copy_from_slice(path);
+
+        Ok(BIP44Path(path_array))
+    }
+
+    pub fn from_string(path: &str) -> Result<BIP44Path, BIP44PathError> {
+        let mut path = path.split('/');
+
+        if path.next() != Some("m") {
+            return Err(BIP44PathError::MissingPrefix);
+        };
+
+        let result = path
+            .map(|index| {
+                let (index_to_parse, mask) = if index.ends_with('\'') {
+                    // Remove the last character and harden index
+                    (&index[..index.len() - 1], HARDENED_BIT)
+                } else {
+                    (index, 0)
+                };
+
+                // FIX ME
```

Comment:
Should we create an issue for this? I am actually not sure about the context or what needs to be fixed.

Reply:
Me neither. I don't see anything that need to be fix

Reply:
I think it was because I was using `.unwrap()` befause and not catching `ParseIntError` which I have done after and just forget to remove the `// FIX ME` message... At least I hope it is that